File: backend/Crystal.Core/Entities/PayStub.cs
namespace Crystal.Core.Entities;

public class PayStub
{
    public int Id { get; set; }

    public int PayPeriodId { get; set; }
    public PayPeriod PayPeriod { get; set; } = null!;

    public int EmployeeProfileId { get; set; }
    public EmployeeProfile EmployeeProfile { get; set; } = null!;

    public int? TimesheetId { get; set; }
    public Timesheet? Timesheet { get; set; }

    public decimal TotalHours { get; set; }
    public decimal GrossPay { get; set; }
    public bool IsPublished { get; set; } = false;

    public bool IsDeleted { get; set; } = false;
}

File: frontend/src/data/types/hr/payStub.ts
export interface PayStubResponseDto {
    id: number;
    payPeriodId: number;
    employeeProfileId: number;
    employeeFirstName: string;
    employeeLastName: string;
    periodStartDate: string;
    periodEndDate: string;
    totalHours: number;
    grossPay: number;
    isPublished: boolean;
}

export interface PayStub extends PayStubResponseDto {
    isDeleted: boolean;
}

export interface GeneratePayrollRequest {
    payPeriodId: number;
    employeeProfileId: number;
}

export interface GeneratePayrollForPeriodRequest {
    payPeriodId: number;
    locationId?: number | null;
}

export interface GeneratePayrollForPeriodResponseDto {
    payPeriodId: number;
    periodStartDate: string;
    periodEndDate: string;
    locationId?: number | null;
    createdCount: number;
    existingCount: number;
    skippedCount: number;
    payStubs: PayStubResponseDto[];
}

export interface GeneratePayrollForPeriodResult extends GeneratePayrollForPeriodResponseDto {
    payStubs: PayStub[];
}
